### DISCUSSION TITLE
Hi there, guys

### DISCUSSION THREAD

**Discussion by adsick** [GraphiteEditor/Graphite#24] [Ideas/Feedback]
I've read the goal of this project and I've found it very similar to what I want to make myself. Currently I am learning Rust. Now I am writing a desktop app which operates a database using Egui and rusqlite, things are going pretty good taking into account that I started learning Rust this winter.
Know what? I like and respect your ambitions, but it seems too much to make once and from scratch, so my point is just to split the project into different parts and then stick to something that can glue them together. Imho that thing could be Bevy engine and the heavy point again is that Bevy is dreaming about it's own editor thing and I found, that integration of Graphite into future Bevy editor will be huge benefit for both. E.g. I really lacked raster/vector graphical editor in Godot engine. Also it would be great if architechture would allow different script plugins of Bevy to work with Graphite so you won't stick to one and only one language.

Hope you don't get angry from my words, best wishes, your Adsick.

**Comment by Keavon** [comment]
Hello and thanks for stopping by! I am glad to hear you are interested in building something very similar to Graphite, and I hope my several years of UX/workflow design can help us both build the best application possible. You are very right in your observation that we have to split it up into chunks and solve small problems one at a time (or in parallel if we have enough interested developers). I will be updating this repository with a set of new issues reflecting the current tasks and roadmap, but I believe it would be good to start a team building the render engine which will become the biggest focus after the GUI is mostly complete. Bevy is doing a lot of awesome stuff very similar to Graphite, although I don't think a shared codebase would be the most effective route forward. However we will definitely want to learn from each other's solutions to problems and eventually integrate our file formats.
